Stakeholders and expectations (#6)
wmde/wikidata-wikibase-architecture: 01-Introduction and Goals.md

## Quality Goals

## Stakeholders & Expectations

### Wikidata Product Manager

- Store structured data for Wikipedia / Wikimedia projects in a central place (Initially sitelinks and infoboxes)

> - It should be possible to create lists from the stored structured data.

Reviewer: ```suggestion
- Allow adding, viewing and editing stored data via both web interface and API
- It should be possible to create lists from the stored structured data.
```

Reviewer: I think the "programmatic access" is important enough to be mentioned separately and has been from the very start.

Author: I am not against this, however is this now not already covered by the "wikidata editor" steakeholder?

> - Ability to add new structured entity data on Wikidata via a desktop web interface.
> - Ability to edit existing structured entity data on Wikidata via a desktop web interface.
> - Ability to programmatically add new structured data to Wikidata.
> - Ability to programmatically edit existing structured data on Wikidata.

Reviewer: yes

- Make the centralized data not only useful for Wikimedia projects, but also other users (researchers, apps etc.)
- Store structured data relevant to the Wiktionary project in a meaningful way, that will also be useful for Abstract Wikipedia.

> - Store structured data relevant to files on Commons.

Reviewer 2: This bullet looks a little out of place to me because everything else applies to Wikidata, but this one is Wikibase-ish functionality intended for Commons. Is this still an expectation by the Wikidata PM?

Author: Very true.
In reality seemingly the commons usecase and all of the architecture changes that came with it were all driven from the Wikidata / Wikimedia side of things (rather than Wikibase). But perhaps this should be moved :)

Author: Done in https://github.com/wmde/wikidata-wikibase-architecture/pull/6/commits/9a7728939ffceae051360bccbbc2b80d2a40fd33

> - The stored data and changes must be managable by a distributed community.

Reviewer 2: ```suggestion
- The stored data and changes must be manageable by a distributed community.
```

- Solution should not enforce too tight rigidity in the data (e.g. enforcing and supporting by automation symmetric properties) because “world is complex and there are always exceptions”

### Wikibase Product Manager

> - People outside of WIkimedia are able to create knowledge bases similar to Wikidata using our same software stack.

Reviewer 2: ```suggestion
- People outside of Wikimedia are able to create knowledge bases similar to Wikidata using our same software stack.
```

- It is easy to write tools and scripts that work for both Wikidata and other Wikibases.
- I can replicate the Wikidata experience on a non Wikimedia Wikibase.

> - I can used the same Wikidata ontology on a non Wikimedia Wikibase.

Reviewer: ```suggestion
- I can use the same Wikidata ontology on a non Wikimedia Wikibase.
- I can easily import my own large data set into a non Wikimedia Wikibase.
```

- Install, configure, update, and maintain a Wikibase instance with confidence.
- Import a large data set into a non Wikimedia Wikibase.
- Use the same Wikidata ontology on a non Wikimedia Wikibase.
- Extend a non Wikimedia Wikibase with relevant entities from Wikidata (federation).

#### Large Non Wikimedia Wikibase Project

- Import large amount of data (>1M entities) into the Wikibase quickly & efficiently.

### Product Managers

- Access to an appropriate test system for product verification for new features.

#### Wikidata Product Manager

- It should be possible to create lists from the stored structured data.
- Make the centralized data not only useful for Wikimedia projects, but also other users (researchers, apps etc.)
- Store structured data relevant to the Wiktionary project in a meaningful way, that will also be useful for Abstract Wikipedia.
- The stored data and changes must be manageable by a distributed community.
- Solution should not enforce too tight rigidity in the data (e.g. enforcing and supporting by automation symmetric properties) because “world is complex and there are always exceptions”

> ### Wikibase / Wikidata System Developers

Reviewer 3: should this be a subsection of the general Product Managers section? so `####`

Author: I dont think so!
